gigascience/gigadb-website
# Fix asset-packagist.org time out with Yii2 composer dependencies

# Pull request for issue: #1024

This is a pull request for the following functionalities:

when running composer install or composer update,
manually or on CI, there is a time out error on https://asset-packagist.org/ [1]

In addition to the need of replacing`` asset-packagist.org`` with ``repo.packagist.org`` in ``composer.json`` [2],
we need to install ``yidas/yii2-composer-bower-skip`` composer library so Yii2 dependencies install/update don't throw errors (which is ok for us as we don't use that part of Yii2)

See: 

[1] hiqdev/asset-packagist/issues/146
[2] https://getcomposer.org/doc/05-repositories.md#composer

## How to test

Running the following commands should not yield any errors:

```
$ docker-compose run --rm config
$ docker-compose run --rm test composer update 
$ docker-compose exec -T console bash -c 'composer update'
$ docker-compose exec -T console bash -c 'cd /gigadb-apps/worker/file-worker/ && composer update'
````
